pyfarm/jobtypes/core/internals.py

```diff
@@ -825,7 +825,7 @@ def _cleanup_system_temp_dir(self, minimum_age=604800):
         for root, dirs, files in os.walk(tempdir, topdown=True,
                                          followlinks=False):
             # Don't delete our own temp files
-            if "pyfarm" in dirs:
+            if root == tempdir and "pyfarm" in dirs:
```

Comment:
The variable `config.tempdir` should be something like `/tmp/pyfarm/agent` so we could just use that to skip our root I think:

```python
# Don't delete our own temp files (agent or otherwise)
if root == dirname(config.tempdir):
    dirs[:] = []  # Do not iterate over sub directories in this root
    files[:] = []  # Do not iterate over files in this root
```

I think this will be more accurate than just checking for 'pyfarm' in dirs, especially if we the config changes.